src/lbaf/IO/configValidator.py

```diff
@@ -0,0 +1,123 @@
+            Optional("logging_level"): And(str, Use(str.lower), lambda f: f in ("info", "debug", "warning", "error"),
+                                           error="info or debug or warning or error needs to be chosen!"),
+            Optional("terminal_background"): And(str, Use(str.lower), lambda g: g in ("light", "dark"),
+                                                 error="light or dark needs to be chosen!"),
+            Optional("output_dir"): str,
+            Optional("generate_multimedia"): bool
+        })
+        self.__from_data = Schema({"data_stem": str, "phase_id": int})
+        self.__from_samplers = Schema({
+            "n_objects": int,
+            "n_mapped_ranks": int,
+            "communication_degree": int,
+            "time_sampler": {"name": And(str, Use(str.lower), lambda a: a in ("uniform", "lognormal"),
+                                         error="uniform or lognormal needs to be chosen!"),
+                             "parameters": [float, float]},
+            "volume_sampler": {"name": And(str, Use(str.lower), lambda b: b in ("uniform", "lognormal"),
+                                           error="uniform or lognormal needs to be chosen!"),
+                               "parameters": [float, float]}
+        })
+        self.__algorithm = {
+            "InformAndTransfer": Schema(
+                {"name": "InformAndTransfer",
+                 "parameters": {"n_iterations": int, "n_rounds": int, "fanout": int, "order_strategy": And(
+                     str, Use(str.lower), lambda e: e in ("arbitrary", "element_id", "increasing_times",
+                                                          "decreasing_times", "increasing_connectivity",
+                                                          "fewest_migrations", "small_objects"),
+                     error="arbitrary or element_id or increasing_times or decreasing_times or increasing_connectivity"
+                           " or fewest_migrations or small_objects needs to be chosen!"),
+                                "criterion": And(str, lambda f: f in ("Tempered", "StrictLocalizer"),
+                                                 error="Tempered or StrictLocalizer needs to be chosen!"),
+                                "max_objects_per_transfer": int,
+                                "deterministic_transfer": bool}}),
+            "BruteForce": Schema(
+                {"name": "BruteForce",
+                 Optional("parameters"): {"skip_transfer": bool}})
+        }
+        self.__logger = logger
+
+    @staticmethod
+    def is_valid(valid_schema: Schema, schema_to_validate: dict) -> bool:
+        """ Returns True is schema_to_validate is valid with self.valid_schema else False. """
+        is_valid = valid_schema.is_valid(schema_to_validate)
+        return is_valid
+
+    @staticmethod
+    def validate(valid_schema: Schema, schema_to_validate: dict):
+        """ Returns validated schema. """
```

Comment:
No `s` at the end of verb: use subjunctive form for descriptions (e.g., `Return validated schema.`)